backend/src/main/java/com/codefolio/backend/user/UserService.java

```diff
@@ -1,30 +1,60 @@
 package com.codefolio.backend.user;
 
+import com.codefolio.backend.user.githubrepo.ProjectsRepository;
+import com.codefolio.backend.user.githubrepo.Projects;
 import com.codefolio.backend.user.session.UserSessionRepository;
-import com.google.gson.Gson;
+import com.codefolio.backend.user.skills.Skills;
+import com.codefolio.backend.user.skills.SkillsRepository;
+import com.codefolio.backend.user.skills.SkillsType;
+import com.codefolio.backend.user.workhistory.Work;
+import com.codefolio.backend.user.workhistory.WorkRepository;
 import lombok.AllArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.Authentication;
 import org.springframework.stereotype.Service;
 
 import java.security.Principal;
+import java.util.List;
 
 @Service
 @AllArgsConstructor
 public class UserService {
 
     private final UserRepository userRepository;
     private final UserSessionRepository userSessionRepository;
+    private final WorkRepository workRepository;
+    private final ProjectsRepository projectsRepository;
+    private final SkillsRepository skillsRepository;
 
     public ResponseEntity<?> userDetails(Principal principal) {
-        if (principal instanceof Authentication) {
-            Object principalUser = ((Authentication) principal).getPrincipal();
-            if (principalUser instanceof Users user) {
-                Gson gson = new Gson();
-                return ResponseEntity.ok(gson.toJson(user));
-            }
+        if (!(principal instanceof Authentication)) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("User not authenticated");
```

Comment:
I would suggest you to send JSON objects for messages like:

```json
{ message: "User not authenticated" }
```

This way you won't have to change the response when in future we need more details along with the error message in the response body. For example:

```json
{
  "message": "Some error message",
  "internalErrorCode": "SomeNumericCode"
}
```

Reply:
Agreed. Will fix this